src/ucp/wireup/select.c

```diff
@@ -1876,17 +1901,25 @@ ucp_wireup_add_rma_bw_lanes(const ucp_wireup_select_params_t *select_params,
                                 UCP_NULL_LANE, select_ctx, 0);
     }
 
+    memset(&dev_count, 0, sizeof(dev_count));
+    bw_info.local_dev_bitmap          = UINT64_MAX;
+    bw_info.remote_dev_bitmap         = UINT64_MAX;
     bw_info.criteria.title            = "high-bw remote memory access";
     bw_info.criteria.lane_type        = UCP_LANE_TYPE_RMA_BW;
     bw_info.criteria.local_cmpt_flags = 0;
 
     if (context->config.ext.proto_enable &&
         (select_params->address->dst_version > UCP_RELEASE_LEGACY)) {
-        bw_info.max_lanes = UCP_PROTO_MAX_LANES;
+        max_lanes   = UCP_PROTO_MAX_LANES;
+        lanes_batch = context->config.ext.rma_bw_lanes_batch;
     } else {
-        bw_info.max_lanes = context->config.ext.max_rndv_lanes;
+        max_lanes   = lanes_batch = context->config.ext.max_rndv_lanes;
     }
 
+    bw_info.max_lanes = lanes_batch;
+    lanes_batch_rem   = max_lanes % lanes_batch;
+    lanes_batch_num   = (max_lanes / lanes_batch) + (lanes_batch_rem ? 1 : 0);
```

Comment:
this approach of counting is more error-prone,
should be something like:
```
while (max_lanes > 0) {
  num_lanes = ... add_lanes(.., ucs_min(max_lanes, lanes_latch)) ...
  max_lanes -= num_lanes;
}
```
also, may need to modify how added_lanes is used

Reply:
Very good suggestion. Updated accordingly.